libs/logger.py

```
@@ -2,6 +2,25 @@
+	"""A console handler that degrades non-encodable characters safely on Windows."""
+
+	def emit(self, record):
+		try:
+			super().emit(record)
+		except UnicodeEncodeError:
+			try:
+				msg = self.format(record)
+				safe_msg = msg.encode("ascii", errors="replace").decode("ascii")
+				self.stream.write(safe_msg + self.terminator)
+				self.flush()
+			except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)